backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/webservice/schema/ResetPasswordQueryService.kt

```diff
@@ -0,0 +1,27 @@
+package app.ehrenamtskarte.backend.auth.webservice.schema
+
+import app.ehrenamtskarte.backend.auth.database.AdministratorEntity
+import app.ehrenamtskarte.backend.auth.database.Administrators
+import app.ehrenamtskarte.backend.auth.webservice.schema.types.Administrator
+import app.ehrenamtskarte.backend.projects.database.ProjectEntity
+import app.ehrenamtskarte.backend.projects.database.Projects
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.transactions.transaction
+import java.time.LocalDateTime
+
+@Suppress("unused")
+class ResetPasswordQueryService {
+    @GraphQLDescription("Verify password reset link")
+    fun checkPasswordResetLink(project: String, resetKey: String): Administrator {
+        return transaction {
+            val projectId = ProjectEntity.find { Projects.project eq project }.single().id.value
+            val admin = AdministratorEntity
+                .find { Administrators.passwordResetKey eq resetKey and (Administrators.projectId eq projectId) }.single()
+            if (admin.passwordResetKeyExpiry!!.isBefore(LocalDateTime.now())) {
+                throw Exception("Password reset key has expired.")
+            }
+            Administrator.fromDbEntity(admin)
```

Comment:
I would rather not return the administrator given a reset key.
People might try to brute force any reset keys with this endpoint, and if they ever find one (which is -- I admit -- quite unlikely) they have full access to the account.

To solve your "usability" issue (#825), I would suggest to also encode the email in the reset link, e.g.
`/reset-password?email={email}&token={resetKey}` 

Then, in the administration frontend, we can autofill the email field from the URL parameter.

Building on top of that, we can also add the email as input to your `checkPasswordResetLink` query.
Then, we look up the administrator with that specific email, check whether the reset key matches (*we should actually store a hash in the DB instead of the raw key!*) and return TRUE if it was successfull, otherwise FALSE.

Related: https://security.stackexchange.com/questions/86913/should-password-reset-tokens-be-hashed-when-stored-in-a-database

Reply:
hm there are very different opinions in this thread. Is it really necessary to hash a reset key thats only valid for 24h? Well then we can create a separate task because thats not really connected to the link checker

Reply:
https://github.com/digitalfabrik/entitlementcard/issues/887

Reply:
adjusted pw reset link to use SearchParams